docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+| guest state reason                 | hypervisor    | -         | -           |
+| memory balloon current             | hypervisor    | -         | -           |
+| memory balloon maximum             | hypervisor    | -         | -           |
+| virtual CPU count                  | hypervisor    | -         | -           |
+| virtual CPU maximum                | hypervisor    | -         | -           |
+| virtual CPU #x state               | hypervisor    | -         | -           |
+| virtual CPU #x total time          | hypervisor    | -         | -           |
+| network interface count            | hypervisor    | -         | -           |
+| block device count                 | hypervisor    | -         | -           |
+| migration downtime                 | hypervisor    | yes       |             |
+| migration elapsed time             | hypervisor    | yes       |             |
+| migration total data               | hypervisor    | yes       |             |
+| migration processed data           | hypervisor    | yes       |             |
+| migration remaining data           | hypervisor    | yes       |             |
+| migration total memory             | hypervisor    | yes       |             |
+| migration processed memory         | hypervisor    | yes       |             |
+| migration remaining memory         | hypervisor    | yes       |             |
+| migration memory bps               | hypervisor    | yes       |             |
+| migration memory constant [TODO]   | hypervisor    | yes       |             |
+| migration memory compression bytes | hypervisor    | yes       |             |
+| migration dirty rate               | hypervisor    | yes       |             |
+| migration memory iteration         | hypervisor    | yes       |             |
+| guest total memory                 | guest agent   | -         |             |
+| guest free memory                  | guest agent   | -         |             |
+| guest CPU count                    | guest agent   | -         |             |
+| guest free disk space              | guest agent   | -         |             |
+| guest IP address                   | guest agent   | -         |             |
+
+
+**NOTE** the list of guest agent provided metrics is partial, because
+a lot of different metrics can be collected for the guest. The presented
+entries are a subset of what the [QEMU Guest Agent](https://wiki.qemu.org/Features/GuestAgent)
+provides, which is used by virtualization managers like
+[oVirt](http://www.ovirt.org)
+
+From now on, the overlapping metrics  (aka 'system metrics') will be ignored,
+and we will focus on the 'hypervisor' and 'guest agent' metrics.
+
+## collection of VM metrics
+A natural fit for the KubeVirt specific metrics is the [custom metrics](
+https://github.com/Kubernetes/Kubernetes/blob/release-1.2/docs/proposals/custom-metrics.md).
+All the metrics are accessible through the libvirt API.
```

Comment:
There can only be one custom metrics provider per cluster for now, which is going to be a prometheus gathering high value metrics), so keep that in mind

Reply:
OK, thanks for pointing this out

Reply:
Relevant (AFAIU) documentation: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/monitoring_architecture.md

Reply:
@smarterclayton could you please point me to some k8s documentation about this limitation of only one custom metrics provider? I want to learn more (e.g. why there is this constraint, if/we it is planned to be lifted, if any other constraint exist) but I didn't find complete documentation yet. Thanks!